### PR TITLE
[FLINK-31126] Move classes not depending on Flink runtime from flink-ml-core to flink-ml-servable-core

### PR DESCRIPTION
**Pull request by lindong28**
## What is the purpose of the change

Move classes not depending on Flink runtime from flink-ml-core to flink-ml-servable-core. This is needed to minimize the dependencies and memory footprint of the module that contains the servable classes in the future.

## Brief change log

Moved classes not depending on Flink runtime from flink-ml-core to flink-ml-servable-core

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by lindong28** [comment]
@zhipeng93 Can you review this PR?

**Review by Fanoid — COMMENTED** [review]
Thanks for the PR. Looks good to me overall.

Just one minor comment: is it possible to rename `ReadWriteUtilsV2` to another name that is more close to its usage, like `JsonUtils`? Because current `ReadWriteUtilsV2` is much different from previous `ReadWriteUtils`.

**Comment by lindong28** [comment]
@Fanoid Thanks for the review. The PR has been updated with ReadWriteUtilsV2  renamed to JsonUtils. Can you take another look?

**Comment by Fanoid** [comment]
> 

Hi, it seems the changes are not pushed.

**Comment by lindong28** [comment]
@Fanoid Ah my bad. Thanks for the reminder. It is updated now.

**Comment by Fanoid** [comment]
Thanks for your updates. LGTM.

File: flink-ml-servable-core/pom.xml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    <version>2.2-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>flink-ml-servable-core</artifactId>
+  <name>Flink ML : Servable : Core</name>
+
+  <dependencies>
+
+    <dependency>
+      <groupId>org.apache.flink</groupId>
```
Comment (zhipeng93) [inline flink-ml-servable-core/pom.xml:37]: It seems that `flink-ml-servable-core` still relies on `flink-core` and is not consistent with the goal of FLIP-289.
Can you explain why do we move XXTypeInfo into `flink-ml-servable-core`? 

Moreover, there are some classes that do not depend on flink runtime, but not used in `flink-ml-servable-core` (e.g., `MLMetrics`). Can we move it back where it is used?

Comment (Fanoid) [reply flink-ml-servable-core/pom.xml:37]: @lindong28 and I had a discussion about `flink-core` in a previous draft PR [1]. We agreed this change can be put in a follow-up PR.

[1] https://github.com/apache/flink-ml/pull/199#discussion_r1090122715

Comment (zhipeng93) [reply flink-ml-servable-core/pom.xml:37]: Thanks for the explanation. I agree it is a reasonable solution in the long term.

**Comment by zhipeng93** [comment]
Thanks for the PR. LGTM.

**Comment by lindong28** [comment]
Thank you both for the review!

**Pull request merged**